tool/internal/instrument/trampoline.go

```diff
@@ -676,37 +687,81 @@ func (ip *InstrumentPhase) rewriteHookContext() {
 	methodGetRetValBody := findSwitchBlock(methodGetRetVal, 0)
 	idx := 0
 	if ast.HasReceiver(ip.targetFunc) {
-		recvType := ip.targetFunc.Recv.List[0].Type
+		splitRecv := ast.SplitMultiNameFields(ip.targetFunc.Recv)
+		recvType := replaceTypeParamsWithAny(splitRecv.List[0].Type, ip.targetFunc.Type.TypeParams)
 		clause := setParamClause(idx, recvType)
 		methodSetParamBody.List = append(methodSetParamBody.List, clause)
 		clause = getParamClause(idx, recvType)
 		methodGetParamBody.List = append(methodGetParamBody.List, clause)
 		idx++
 	}
-	for _, param := range ip.targetFunc.Type.Params.List {
-		paramType := desugarType(param)
-		for range param.Names {
-			clause := setParamClause(idx, paramType)
-			methodSetParamBody.List = append(methodSetParamBody.List, clause)
-			clause = getParamClause(idx, paramType)
-			methodGetParamBody.List = append(methodGetParamBody.List, clause)
-			idx++
-		}
+	splitParams := ast.SplitMultiNameFields(ip.targetFunc.Type.Params)
+	for _, param := range splitParams.List {
+		paramType := replaceTypeParamsWithAny(desugarType(param), ip.targetFunc.Type.TypeParams)
+		clause := setParamClause(idx, paramType)
+		methodSetParamBody.List = append(methodSetParamBody.List, clause)
+		clause = getParamClause(idx, paramType)
+		methodGetParamBody.List = append(methodGetParamBody.List, clause)
+		idx++
 	}
```

Comment:
The SetParam/SetReturnVal function is intended to modify the parameters of the original function, but the current implementation doesn't allow this. We should rewrite it to panic (or something similar) to reject user calls and make it clear this operation is unsupported.

Reply:
To clarify, what I meant is that we should generate code like the following:

```go
func (c *HookContextImpl3580597051) SetParam/SetReturnVal(idx int, val interface{}) {
	panic("Unsupported method")
}
```

This shouldn't be too difficult, I think. Couldn't we just restore the original code and then add an if branch to handle this case?